airflow/bin/cli.py

```diff
@@ -354,7 +374,8 @@ def run(args, dag=None):
         # via the UI (or vice versa) results in a permission error as the task
         # tries to write to a log file created by the other user.
         log_base = os.path.expanduser(conf.get('core', 'BASE_LOG_FOLDER'))
-        directory = log_base + "/{args.dag_id}/{args.task_id}".format(args=args)
+        log_relative_dir = logging_utils.get_log_directory(**locals())
```

Comment:
I agree, and made a similar comment earlier. We should have an abstraction for logging.